src/Sticky.js

```diff
@@ -85,24 +85,20 @@ export default class Sticky extends Component {
       style
     });
   };
+  
+  contentRef = content => {
+    this.content = ReactDOM.findDOMNode(content);
+  };
+  placeholderRed = placeholder => {
+    this.placeholder = placeholder
+  };
 
   render() {
-    const element = React.cloneElement(
-      this.props.children({
-        isSticky: this.state.isSticky,
-        wasSticky: this.state.wasSticky,
-        distanceFromTop: this.state.distanceFromTop,
-        distanceFromBottom: this.state.distanceFromBottom,
-        calculatedHeight: this.state.calculatedHeight,
-        style: this.state.style
-      }),
-      { ref: content => { this.content = ReactDOM.findDOMNode(content); } }
-    )
-
+    const Component = this.props.children;
     return (
       <div>
-        <div ref={ placeholder => this.placeholder = placeholder } />
-        { element }
+        <div ref={this.placeholderRef} />
+        <Component {...this.state} ref={this.contentRef}/>
```

Comment:
I didn't run anything, I put this in to get your feel for it. I can harden it up. I made these changes to the lib files in my node_modules as a POC and then just sent it over via edit in GH.